saltproc/saltproc.py

```diff
@@ -265,108 +500,166 @@ def write_mat_file(self):
+                      (self.removed_qty, group, scheme['from']))
+            else:
+                continue
+
+        # get mass lacking in reactor after reprocessing out waste
+        self.core_space = {}
+        for mat, val in self.core.items():
+            self.core_space[mat] = self.core_mass[mat] - sum(val)
+
+        # move things around (fissile to driver)
+        for group, scheme in self.rep_scheme.items():
+            if scheme['to'] != 'waste' and scheme['from'] != 'fertile':
+                removed = self.remove_iso(
+                    iso_indx, scheme['eff'], scheme['from'])
+                print('REMOVING %f kg of %s FROM %s' %
+                      (self.removed_qty, group, scheme['from']))
+                # if the movement flow is more than the space in the destination,
+                if sum(removed) > self.core_space[scheme['to']]:
+                    removed_comp = removed / sum(removed)
+                    self.core[scheme['to']] += removed_comp * \
+                        self.core_space[scheme['to']]
+                    print('MOVING %f kg of %s FROM %s TO %s ' % (
+                        self.core_space[scheme['to']], group, scheme['from'], scheme['to']))
+                    # move rest to fissile tank
+                    self.fissile_tank_db[self.current_step, :] += removed_comp * \
+                        (sum(removed) - self.core_space[scheme['to']])
+                    print('MOVING %f kg of %s FROM %s TO FISSILE TANK' %
+                          (self.core_space[scheme['to']], group, scheme['from']))
+
+    def refuel(self):
+        """ After separating out fissile and waste material,
+            this function refuels the salt with fissile and fertile
+            material
+        """
+        # refill tank db initialization
+        self.driver_refill_tank_db[self.current_step,
+                                   :] = self.driver_refill_tank_db[self.current_step-1, :]
+        self.blanket_refill_tank_db[self.current_step,
+                                    :] = self.blanket_refill_tank_db[self.current_step-1, :]
+
+        for group, scheme in self.rep_scheme.items():
+            if scheme['from'] != 'fertile':
+                continue
+            else:
+                qty_to_fill = self.core_mass[scheme['to']
+                                             ] - sum(self.core[scheme['to']])
+                for indx, frac in enumerate(scheme['comp']):
+                    isoid = self.find_iso_indx(scheme['element'][indx])
+                    self.refill(isoid, qty_to_fill*frac, scheme['to'])
+                    print('ADDING IN %f kg of %s to %s' % (
+                        qty_to_fill * frac, scheme['element'][indx], scheme['to']))
+
+    def reactivity_control(self):
+        """ Controls fraction of fissile material
+            input into core to control keff into
+            a range
+        """
+        # check EOC KEFF for determining fissile_add_back_frac
```

Comment:
comment should be deleted.
also, random blank line below should be deleted.
don't begin functions with blank lines.